blockstore/src/sled_blockstore.rs

```diff
@@ -0,0 +1,220 @@
+use std::io;
+use std::sync::Arc;
+
+use cid::CidGeneric;
+use sled::{Db, Error as SledError, Tree};
+use tokio::task::spawn_blocking;
+use tokio::task::JoinError;
+
+use crate::{convert_cid, Blockstore, BlockstoreError, Result};
+
+const BLOCKS_TREE_ID: &[u8] = b"BLOCKSTORE.BLOCKS";
+
+/// A [`Blockstore`] implementation backed by a [`sled`] database.
+#[derive(Debug)]
+pub struct SledBlockstore<const MAX_MULTIHASH_SIZE: usize> {
+    inner: Arc<Inner>,
+}
+
+#[derive(Debug)]
+struct Inner {
+    _db: Db,
+    blocks: Tree,
+}
+
+impl<const MAX_MULTIHASH_SIZE: usize> SledBlockstore<MAX_MULTIHASH_SIZE> {
+    /// Create or open a [`SledBlockstore`] in a given sled [`Db`].
+    ///
+    /// # Example
+    /// ```
+    /// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+    /// use blockstore::SledBlockstore;
+    /// use tokio::task::spawn_blocking;
+    ///
+    /// let db = spawn_blocking(|| sled::open("path/to/db")).await??;
+    /// let blockstore = SledBlockstore::<64>::new(db).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn new(db: Db) -> Result<Self> {
+        let blockstore = spawn_blocking(move || {
+            let blocks = db.open_tree(BLOCKS_TREE_ID)?;
+
+            Ok::<_, SledError>(Self {
+                inner: Arc::new(Inner { _db: db, blocks }),
+            })
+        })
+        .await??;
+        Ok(blockstore)
```

Comment:
```suggestion
        spawn_blocking(move || {
            let blocks = db.open_tree(BLOCKS_TREE_ID)?;

            Ok(Self {
                inner: Arc::new(Inner { _db: db, blocks }),
            })
        })
        .await?
```
gets rid of manually specifying Err type, wdyt?